tests/js/common/shell/shell-collection-api.js

```diff
@@ -0,0 +1,1691 @@
+/*jshint globalstrict:false, strict:false */
+/*global assertTrue, assertEqual, assertFalse, assertNotEqual, fail */
+
+////////////////////////////////////////////////////////////////////////////////
+/// @brief test the authentication
+///
+/// DISCLAIMER
+///
+/// Copyright 2023-2023 ArangoDB GmbH, Cologne, Germany
+///
+/// Licensed under the Apache License, Version 2.0 (the "License");
+/// you may not use this file except in compliance with the License.
+/// You may obtain a copy of the License at
+///
+///     http://www.apache.org/licenses/LICENSE-2.0
+///
+/// Unless required by applicable law or agreed to in writing, software
+/// distributed under the License is distributed on an "AS IS" BASIS,
+/// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+/// See the License for the specific language governing permissions and
+/// limitations under the License.
+///
+/// Copyright holder is ArangoDB GmbH, Cologne, Germany
+///
+/// @author Copyright 2023, ArangoDB GmbH, Cologne, Germany
+////////////////////////////////////////////////////////////////////////////////
+
+const jsunity = require("jsunity");
+const {arango, db, ArangoError} = require("@arangodb");
+const isCluster = require('internal').isCluster();
+const isEnterprise = require("internal").isEnterprise();
+const _ = require("lodash");
+const isWindows = (require("internal").platform.substr(0, 3) === 'win');
+const isServer = typeof require("internal").arango === 'undefined';
+
+const {
+  ERROR_HTTP_BAD_PARAMETER,
+  ERROR_HTTP_SERVER_ERROR,
+  ERROR_BAD_PARAMETER,
+  ERROR_VALIDATION_BAD_PARAMETER,
+  ERROR_INVALID_SMART_JOIN_ATTRIBUTE,
+  ERROR_ARANGO_COLLECTION_TYPE_INVALID,
+  ERROR_CLUSTER_INSUFFICIENT_DBSERVERS,
+  ERROR_HTTP_CONFLICT,
+  ERROR_ARANGO_DUPLICATE_NAME,
+  ERROR_CLUSTER_UNKNOWN_DISTRIBUTESHARDSLIKE,
+  ERROR_INTERNAL,
+  ERROR_ARANGO_ILLEGAL_NAME,
+  ERROR_HTTP_CORRUPTED_JSON
+} = require("internal").errors;
+
+const filterClusterOnlyAttributes = (attributes) => {
+  // The following attributes are not returned by the Collection Properties API
+  // However the user can still send them to the API. Erase them here as they should
+  // not be returned by the server.
+  if (!isCluster) {
+    const clusterOnlyAttributes = ["numberOfShards", "shardKeys", "replicationFactor"];
+    /* , , "writeConcern" */
```

Comment:
why is `writeConcern` not part of this list? It should be a cluster-only attribute. If it's not, please leave at least a comment in the test file why it is not included in the list.
Wondering also if we should exclude `smartJoinAttribute` here.

Reply:
We may also still return the `minReplicationFactor` attribute, which was the predecessor of `writeConcern`. Should it be excluded here too?

Reply:
Same for `shardingStrategy`, which should be cluster-only.

Reply:
1) writeConcern is actually exposed on SingleServer, therefore it is not on the list.
Will add a comment.
2) minReplicationFactor is sometimes exposed on SingleServer, namely when this filtering is disabled
2a) Enterprise Edition using Enterprise Features like satellites and SmartGraphs disable this filtering, they expose the given list of ClusterOnlyAttributes even on SingleServer.
3) shardingStrategy and distributeShardsLike seem to be always ignored.

I try to come up with a way to make this a bit easier to understand on the Test side.